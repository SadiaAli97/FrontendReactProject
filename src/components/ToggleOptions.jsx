import * as React from 'react';
import ToggleButton from '@mui/material/ToggleButton';
import ToggleButtonGroup from '@mui/material/ToggleButtonGroup';

export function ToggleOptions({selectedToggle, toggleChange}) {
  const [alignment, setAlignment] = React.useState('isdks');

  const handleChange = (event, newAlignment) => {
    setAlignment(newAlignment);
    toggleChange(newAlignment);
  };

  return (
    <ToggleButtonGroup
      color="primary"
      value={alignment}
      exclusive
      onChange={handleChange}
    >
      <ToggleButton value="isdks">Installed</ToggleButton>
      <ToggleButton value="uisdks">Uninstalled</ToggleButton>
    </ToggleButtonGroup>
  );
}