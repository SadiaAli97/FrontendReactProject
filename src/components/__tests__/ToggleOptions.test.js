import TestRenderer from 'react-test-renderer';
import { ToggleOptions } from '../ToggleOptions';

it('renders ToggleOptions ', () => {
	const testRenderer = TestRenderer.create(
		<ToggleOptions
		  selectedToggle='isdks'
		  toggleChange={jest.fn()}
		/>
	);
  expect(testRenderer.toJSON()).toMatchSnapshot()
});

