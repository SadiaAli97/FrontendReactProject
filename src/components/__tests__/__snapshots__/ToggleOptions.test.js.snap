// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ToggleOptions  1`] = `
<div
  className="MuiToggleButtonGroup-root css-1gjgmky-MuiToggleButtonGroup-root"
  role="group"
>
  <button
    aria-pressed={true}
    className="MuiButtonBase-root MuiToggleButton-root Mui-selected MuiToggleButton-sizeMedium MuiToggleButton-primary MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal css-d9c359-MuiButtonBase-root-MuiToggleButton-root"
    disabled={false}
    onBlur={[Function]}
    onChange={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
    value="isdks"
  >
    Installed
  </button>
  <button
    aria-pressed={false}
    className="MuiButtonBase-root MuiToggleButton-root MuiToggleButton-sizeMedium MuiToggleButton-primary MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal css-d9c359-MuiButtonBase-root-MuiToggleButton-root"
    disabled={false}
    onBlur={[Function]}
    onChange={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
    value="uisdks"
  >
    Uninstalled
  </button>
</div>
`;
